framework/Distributions.py

```diff
@@ -1884,6 +1896,18 @@ def initializeDistribution(self):
     initialPerm = randomUtils.randomPermutation(self.xArray.tolist(),self)
     self.pot = np.asarray(initialPerm)
 
+  def initializeFromDict(self, inputDict):
+    """
+      Function that initializes the distribution provided a dictionary
+      @ In, inputDict, dict, dictionary containing the np.arrays for xAxis and pAxis
+      @ Out, None
+    """
+    self.strategy = inputDict['strategy']
+    self.categoricalDist = Categorical()
+    self.categoricalDist.initializeFromDict(inputDict)
+    initialPerm = randomUtils.randomPermutation(inputDict['xAxis'].tolist(),self)
+    self.pot = np.asarray(initialPerm)
```

Comment:
please try to define self.strategy, self.categoricalDist, and self.pot in the __init__ method, and provide the descriptions for each self variables.

Reply:
description added